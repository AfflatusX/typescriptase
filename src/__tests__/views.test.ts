/**
 * This file is partially generated; only edit bespoke sections.
 *
 * SOURCE<<gen/tests.ts::viewsTest>>
 * BESPOKE<<imports, beforeAll, afterAll, beforeEach, afterEach, simpleView, propsView, stateView, fragmentRelayView, paginationRelayView, refetchRelayView>>
 * SIGNED<<iN9XEf7i1TWbU0KPmPX6rERE/SvM/n74FJOVkqnV5Mhmqf5hyPGndY0N1dsAHeKYZ/Rmb8f6kY6x6vsULoGvkA==>>
 */

/* BESPOKE START <<imports>> */
import {
  fragmentRelayView,
  paginationRelayView,
  propsView,
  refetchRelayView,
  simpleView,
  stateView,
} from "../../gen/views";
/* BESPOKE END <<imports>> */

beforeAll(
  async (): Promise<void> => {
    /* BESPOKE START <<beforeAll>> */
    /* BESPOKE END <<beforeAll>> */
  },
);

afterAll(
  async (): Promise<void> => {
    /* BESPOKE START <<afterAll>> */
    /* BESPOKE END <<afterAll>> */
  },
);

beforeEach(
  async (): Promise<void> => {
    /* BESPOKE START <<beforeEach>> */
    /* BESPOKE END <<beforeEach>> */
  },
);

afterEach(
  async (): Promise<void> => {
    /* BESPOKE START <<afterEach>> */
    /* BESPOKE END <<afterEach>> */
  },
);

test(
  "simpleView",
  async (): Promise<void> => {
    /* BESPOKE START <<simpleView>> */
    expect(simpleView.print())
      .toMatchSnapshot();
    /* BESPOKE END <<simpleView>> */
  },
);

test(
  "propsView",
  async (): Promise<void> => {
    /* BESPOKE START <<propsView>> */
    expect(propsView.print())
      .toMatchSnapshot();
    /* BESPOKE END <<propsView>> */
  },
);

test(
  "stateView",
  async (): Promise<void> => {
    /* BESPOKE START <<stateView>> */
    expect(stateView.print())
      .toMatchSnapshot();
    /* BESPOKE END <<stateView>> */
  },
);

test(
  "fragmentRelayView",
  async (): Promise<void> => {
    /* BESPOKE START <<fragmentRelayView>> */
    expect(fragmentRelayView.print())
      .toMatchSnapshot();
    /* BESPOKE END <<fragmentRelayView>> */
  },
);

test(
  "paginationRelayView",
  async (): Promise<void> => {
    /* BESPOKE START <<paginationRelayView>> */
    expect(paginationRelayView.print())
      .toMatchSnapshot();
    /* BESPOKE END <<paginationRelayView>> */
  },
);

test(
  "refetchRelayView",
  async (): Promise<void> => {
    /* BESPOKE START <<refetchRelayView>> */
    expect(refetchRelayView.print())
      .toMatchSnapshot();
    /* BESPOKE END <<refetchRelayView>> */
  },
);
