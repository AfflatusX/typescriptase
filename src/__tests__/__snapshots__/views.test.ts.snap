// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fragmentRelayView 1`] = `
"/**
 * This file is partially generated; only edit bespoke sections.
 *
 * SOURCE<<DO_NOT_COMMIT::DO_NOT_COMMIT>>
 * BESPOKE<<imports, state, render, implementation, relay>>
 * SIGNED<<KCcRsWvI5VURL5ffx/hmd1wEjLt+Sikg5O2j2Pg2ModZl8inTRK8oijE91X/JyzUxuQOGa0krP0di9/zGEt6QQ==>>
 */

import * as React from \\"react\\";
import {
  polyfill,
} from \\"react-lifecycles-compat\\";
import {
  createFragmentContainer,
  graphql,
} from \\"react-relay\\";

/* BESPOKE START <<imports>> */
/* BESPOKE END <<imports>> */

export interface IFragmentRelayProps {
  test1: string;
}

export interface IFragmentRelayState {
  test2?: string;
}

class FragmentRelayImpl extends React.Component<IFragmentRelayProps, IFragmentRelayState> {

  public constructor(
    props: IFragmentRelayProps,
  ) {
    super(props);
    this.state = {
      /* BESPOKE START <<state>> */
      /* BESPOKE END <<state>> */
    };
  }

  public render(
  ): JSX.Element {
    /* BESPOKE START <<render>> */
    /* BESPOKE END <<render>> */
  }

  /* BESPOKE START <<implementation>> */
  /* BESPOKE END <<implementation>> */
}

const FragmentRelay: React.ComponentType = createFragmentContainer(
  FragmentRelayImpl,
  /* BESPOKE START <<relay>> */
  /* BESPOKE END <<relay>> */
);
"
`;

exports[`paginationRelayView 1`] = `
"/**
 * This file is partially generated; only edit bespoke sections.
 *
 * SOURCE<<DO_NOT_COMMIT::DO_NOT_COMMIT>>
 * BESPOKE<<imports, render, implementation, relay>>
 * SIGNED<<8Q0bO2m20bd4r2AOvYYcQHTPGRmWYEtJcTC+frKR50VpLcLcKog8O6m1M3VF2rb15QTFRzbarWRwyU2u5JNsVQ==>>
 */

import * as React from \\"react\\";
import {
  polyfill,
} from \\"react-lifecycles-compat\\";
import {
  createPaginationContainer,
  graphql,
} from \\"react-relay\\";

/* BESPOKE START <<imports>> */
/* BESPOKE END <<imports>> */

export interface IPaginationRelayProps {
  test1: string;
}

class PaginationRelayImpl extends React.Component<IPaginationRelayProps> {

  public render(
  ): JSX.Element {
    /* BESPOKE START <<render>> */
    /* BESPOKE END <<render>> */
  }

  /* BESPOKE START <<implementation>> */
  /* BESPOKE END <<implementation>> */
}

const PaginationRelay: React.ComponentType = createPaginationContainer(
  PaginationRelayImpl,
  /* BESPOKE START <<relay>> */
  /* BESPOKE END <<relay>> */
);
"
`;

exports[`propsView 1`] = `
"/**
 * This file is partially generated; only edit bespoke sections.
 *
 * SOURCE<<DO_NOT_COMMIT::DO_NOT_COMMIT>>
 * BESPOKE<<imports, render, implementation>>
 * SIGNED<</x/CRORF5gap0d9piiEhMcr8RYXIXaMTry45H3iESbRtbvAFrSbTDR3x/DYrVtaLsDNkIfXa2555Z+CuV9cG1A==>>
 */

import * as React from \\"react\\";
import {
  polyfill,
} from \\"react-lifecycles-compat\\";

/* BESPOKE START <<imports>> */
/* BESPOKE END <<imports>> */

export interface IPropsProps {
  test1: string;
}

class Props extends React.Component<IPropsProps> {

  public render(
  ): JSX.Element {
    /* BESPOKE START <<render>> */
    /* BESPOKE END <<render>> */
  }

  /* BESPOKE START <<implementation>> */
  /* BESPOKE END <<implementation>> */
}
"
`;

exports[`refetchRelayView 1`] = `
"/**
 * This file is partially generated; only edit bespoke sections.
 *
 * SOURCE<<DO_NOT_COMMIT::DO_NOT_COMMIT>>
 * BESPOKE<<imports, state, render, implementation, relay>>
 * SIGNED<<KzdICieyngONMGB4wSWKjEm6orIPuWjHW7+iPPZsw23fIBh4w27OODupZKnokNvGhz94c3/wAtte0JkauPgqWA==>>
 */

import * as React from \\"react\\";
import {
  polyfill,
} from \\"react-lifecycles-compat\\";
import {
  commitMutation,
  createRefetchContainer,
  graphql,
} from \\"react-relay\\";

/* BESPOKE START <<imports>> */
/* BESPOKE END <<imports>> */

export interface IRefetchRelayProps {
  test1: string;
}

export interface IRefetchRelayState {
  test2?: string;
}

class RefetchRelayImpl extends React.Component<IRefetchRelayProps, IRefetchRelayState> {

  public constructor(
    props: IRefetchRelayProps,
  ) {
    super(props);
    this.state = {
      /* BESPOKE START <<state>> */
      /* BESPOKE END <<state>> */
    };
  }

  public render(
  ): JSX.Element {
    /* BESPOKE START <<render>> */
    /* BESPOKE END <<render>> */
  }

  /* BESPOKE START <<implementation>> */
  /* BESPOKE END <<implementation>> */
}

const RefetchRelay: React.ComponentType = createRefetchContainer(
  RefetchRelayImpl,
  /* BESPOKE START <<relay>> */
  /* BESPOKE END <<relay>> */
);
"
`;

exports[`simpleView 1`] = `
"/**
 * This file is partially generated; only edit bespoke sections.
 *
 * SOURCE<<DO_NOT_COMMIT::DO_NOT_COMMIT>>
 * BESPOKE<<imports, render>>
 * SIGNED<<3a88lFatMjcrEeOvQGAmE/7xXhD6ihh/YpQ7RH/UQU82MXLmM4/qDSJI6Heu13H4YhRija7dsd9KspXjxKUIVA==>>
 */

import * as React from \\"react\\";
import {
  polyfill,
} from \\"react-lifecycles-compat\\";

/* BESPOKE START <<imports>> */
/* BESPOKE END <<imports>> */

export function Simple(
): JSX.Element {
  /* BESPOKE START <<render>> */
  /* BESPOKE END <<render>> */
}
"
`;

exports[`stateView 1`] = `
"/**
 * This file is partially generated; only edit bespoke sections.
 *
 * SOURCE<<DO_NOT_COMMIT::DO_NOT_COMMIT>>
 * BESPOKE<<imports, state, render, implementation>>
 * SIGNED<<gvm5/oglmJO7PRafOM/YHb/8XHHk4GFnAq9itdomnyVzrwXhVUULeBgjwBjDpO2lRS0htuh+mLhSXyGn/yYk8g==>>
 */

import * as React from \\"react\\";
import {
  polyfill,
} from \\"react-lifecycles-compat\\";
import {
  commitMutation,
  graphql,
} from \\"react-relay\\";

/* BESPOKE START <<imports>> */
/* BESPOKE END <<imports>> */

export interface IStateProps {
  test1: string;
}

export interface IStateState {
  test2?: string;
}

class State extends React.Component<IStateProps, IStateState> {

  public constructor(
    props: IStateProps,
  ) {
    super(props);
    this.state = {
      /* BESPOKE START <<state>> */
      /* BESPOKE END <<state>> */
    };
  }

  public render(
  ): JSX.Element {
    /* BESPOKE START <<render>> */
    /* BESPOKE END <<render>> */
  }

  /* BESPOKE START <<implementation>> */
  /* BESPOKE END <<implementation>> */
}
"
`;
